interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -233,14 +234,17 @@ private void startServer() {
                 new TestServiceImpl(testServiceExecutor),
                 allInterceptors))
         .addStreamTracerFactory(serverStreamTracerFactory);
-    io.grpc.internal.TestingAccessor.setStatsImplementation(
-        builder,
-        new CensusStatsModule(
-            tagger,
-            tagContextBinarySerializer,
-            serverStatsRecorder,
-            GrpcUtil.STOPWATCH_SUPPLIER,
-            true, true, true, false /* real-time metrics */));
+    if (metricsExpected()) {
```

Comment:
Even if metrics are not expected, census context propagation may still happen, e.g., in `InProcessTest`. You need to inject fake Census whenever builder is `AbstractServerImplBuilder`, regardless of `metricsExpected()`, and then have the assert:
```java
if (metricsExpected()) {
  assertThat(builder).isInstanceOf(AbstractServerImplBuilder.class).
}
```

Reply:
I changed it to just check the class.  PTAL

Reply:
Please add the assert I suggested. It's much easier to debug this assert failing than metrics not being recorded.

Reply:
Done, (sorry, I didn't understand why you wanted it at first).